src/ui/components/__tests__/AppModal.test.tsx

```diff
@@ -1,57 +1,56 @@
 import userEvent from "@testing-library/user-event";
 import React from "react";
-import { act } from "react-dom/test-utils";
 import { loadFixtureData } from "test/testFixtureUtils";
 import { render, filterCommonTestWarnings, screen } from "test/testUtils";
 import { UIStore } from "ui/actions";
 import { setLoadingFinished, setModal } from "ui/actions/app";
+import { Recording } from "ui/types";
+import { getRecordingURL } from "ui/utils/recording";
 
 const useRouter = jest.spyOn(require("next/router"), "useRouter");
 
-const FAKE_RECORDING_ID = "mock-recording";
-
-useRouter.mockImplementation(() => ({
-  asPath: `/recording/${FAKE_RECORDING_ID}`,
-  query: { id: FAKE_RECORDING_ID },
-}));
-
 describe("AppModal", () => {
   filterCommonTestWarnings();
 
+  let graphqlMocks: any[] = null as unknown as any[];
+  let recording: Recording = null as unknown as Recording;
   let store: UIStore = null as unknown as UIStore;
 
   beforeEach(async () => {
-    store = await loadFixtureData("console_messages");
+    const response = await loadFixtureData("console_messages");
+    graphqlMocks = response.graphqlMocks;
+    recording = response.recording;
+    store = response.store;
+
+    const url = getRecordingURL(recording);
+    global.location.pathname = url;
+    useRouter.mockImplementation(() => ({
+      asPath: url,
+      pathname: url,
+      query: { id: recording.id },
+    }));
```

Comment:
This block is probably something we should move to test utils at some point 😄  A few tests are doing this.